app/packages/core/src/components/SamplesContainer.tsx

```diff
@@ -134,27 +138,28 @@ function SamplesContainer() {
     []
   );
 
-  const defaultSpacesState = {
-    id: "root",
-    children: [
-      {
-        id: "default-samples-node",
-        children: [],
-        type: "Samples",
-        pinned: true,
-      },
-    ],
-    type: "panel-container",
-    activeChild: "default-samples-node",
-  };
+  useEffect(() => {
+    if (!spaces.equals(sessionSpaces)) {
+      updateSpaces(sessionSpaces);
+    }
+  }, [sessionSpaces]);
+
+  useEffect(() => {
+    if (!isMounted.current) {
+      isMounted.current = true;
+      return;
+    }
+    const serializedSpaces = spaces.toJSON();
+    if (!spaces.equals(sessionSpaces) && !spaces.equals(oldSpaces.current)) {
+      setSessionSpaces(serializedSpaces);
+    }
+    oldSpaces.current = serializedSpaces;
+  }, [spaces]);
 
   return (
     <Container>
-      {showSidebar && <Sidebar render={renderGridEntry} modal={false} />}
-      <SpacesRoot
-        id={`primary-${datasetName}`}
-        defaultState={defaultSpacesState}
-      />
+      {showSidebar && <Sidebar render={renderGridEntry} modal={false} />}{" "}
```

Comment:
Assuming the quotes were unintentional
```suggestion
      {showSidebar && <Sidebar render={renderGridEntry} modal={false} />
```

Reply:
Yah, I believe formatter likely added those. Will remove